### ISSUE TITLE
Actions from actionQueue are sometimes dispatched multiple times

### ISSUE THREAD

**Issue by juspeeh** [rgommezz/react-native-offline#344]
### Current Behavior

- What code are you running and what is happening?

Actions from actionQueue are sometimes dispatched multiple times

### Expected Behavior

- What do you expect should be happening?

Every action from actionQueue should be dispatched only once

### How to reproduce
When "@@network-connectivity/CONNECTION_CHANGE" action queue to be flushed is created. If action inside that queue dispatches "@@network-connectivity/CHANGE_QUEUE_SEMAPHORE", payload "GREEN"( as the readME recommends if you want to wait an action to complete before the next action). Due to the action being intercepted by the library a secondary queue will be created and those both are ran simultaneously. Nothing is done to check if certain action is already removed from the actual state or queue is already being flushed. This causes X amount of actions to be dispatched multiple times depending on the queue size.

**Comment by superyarik** [comment]
facing same issue

**Comment by HamidAliSE** [comment]
I am still facing this issue on react-native 0.79.0. Surprisingly I am also using 2000 as queueReleaseThrottle. When I had a delay of 500, some requests was getting missed. That's why I have changed it to 2000. Now I am stuck with another issue. Any update on this???
